bitcoin_executor/sources/interpreter.move

```diff
@@ -390,6 +398,26 @@ fun op_dup(ip: &mut Interpreter) {
     ip.stack.push(value)
 }
 
+fun op_drop(ip: &mut Interpreter) {
+    assert!(!ip.stack.is_empty(), EInvalidStackOperation);
```

Comment:
When popping from the stack, we check whether the stack is empty or not.

Reply:
So we don't need to check this empty again. If this is empty, this will return an error. You can check this logic in the stack.move

Reply:
I saw we are checking for this and abort. I liked to have this error where it happens. rather than checking if the underlying code will handle it for me. I agree that repeating the condition check is not ideal. Maybe we should rename `pop` to `try_pop` and return and `Option`? and then handle it in the opcodes? Bitcoin-core returns error on the scirpt level

Reply:
created issued for this